backend/testing/tester.py

```diff
@@ -0,0 +1,158 @@
+
+    hijack_queue = Queue(
+        'hijack-testing',
+        exchange=hijack_exchange,
+        routing_key='update',
+        durable=False,
+        auto_delete=True,
+        max_priority=1,
+        consumer_arguments={'x-priority': 1})
+
+    hijack_db_queue = Queue(
+        'hijack-db-testing',
+        exchange=pg_amq_bridge,
+        routing_key='hijack-update',
+        durable=False,
+        auto_delete=True,
+        max_priority=1,
+        consumer_arguments={'x-priority': 1})
+
+    process_state_queue = Queue(
+        'process-state-testing',
+        exchange=pg_amq_bridge,
+        routing_key='process-state',
+        durable=False,
+        auto_delete=True,
+        max_priority=1,
+        consumer_arguments={'x-priority': 1})
+
+    messages = {}
+    with open('messages.json', 'r') as f:
+        messages = json.load(f)
+
+    send_len = len(messages)
+
+    def validate_message(body, message):
+        global curr_idx, expected_messages
+        print('\t- Receiving Batch #{} - Type {} - Remaining {}'.format(curr_idx, message.delivery_info['routing_key'], expected_messages-1))
+        if isinstance(body, dict):
+            event = body
+        else:
+            event = json.loads(body)
+        # logging.debug(event)
+
+        if message.delivery_info['routing_key'] == 'update-insert':
+            expected = messages[curr_idx]['detection_update_response']
+        elif message.delivery_info['routing_key'] == 'update':
+            expected = messages[curr_idx]['detection_hijack_response']
+        elif message.delivery_info['routing_key'] == 'hijack-update':
+            expected = messages[curr_idx]['database_hijack_response']
+
+        for key in set(event.keys()).intersection(expected.keys()):
+            if event[key] != expected[key]:
+                sys.exit('Unexpected value for key \"{}\"\nReceived: {}, Expected: {}'
+                         .format(key, event[key], expected[key]))
+
+        expected_messages -= 1
+        if expected_messages <= 0:
+            curr_idx += 1
+        message.ack()
+
+    def send_next_message(conn):
+        global curr_idx, expected_messages
+        with conn.Producer() as producer:
+            expected_messages = len(messages[curr_idx]) - 1
+            print('Publishing #{}'.format(curr_idx))
+            # logging.debug(messages[curr_idx]['send'])
+
+            producer.publish(
+                messages[curr_idx]['send'],
+                exchange=update_exchange,
+                routing_key='update',
+                serializer='json'
+            )
+
+    with Connection(RABBITMQ_HOST) as connection:
+        pg_amq_bridge.declare(channel=connection.default_channel)
+        bind_queue = process_state_queue(connection.default_channel)
+        bind_queue.declare()
+        processes = 0
+        while True:
+            msg = bind_queue.get()
+            if msg:
+                processes += 1
+                if processes == 13:
```

Comment:
why 13? Please create a constant and add a comment.

Reply:
7 modules initialized and 6 are started afterwards making for 7 inserts and 6 updates (13)